aidial_adapter_bedrock/llm/model/stabililty/v3.py

```diff
@@ -0,0 +1,177 @@
+
+
+class StabilityV3Response(BaseModel):
+    seeds: List[int]
+    images: List[str]
+    # None will indicate that the request was successful
+    # Possible values:
+    # "Filter reason: prompt"
+    # "Filter reason: output image"
+    # "Filter reason: input image"
+    # "Inference error"
+    # null
+    finish_reasons: List[Optional[str]]
+
+    def content(self) -> str:
+        return " "
+
+    def attachments(self) -> List[Attachment]:
+        return [
+            Attachment(
+                type="image/png",
+                data=image,
+            )
+            for image in self.images
+        ]
+
+    def usage(self) -> TokenUsage:
+        return TokenUsage(prompt_tokens=0, completion_tokens=1)
+
+    def throw_if_error(self):
+        error = next((reason for reason in self.finish_reasons if reason), None)
+        if not error:
+            return
+
+        if error == "Inference error":
+            raise RuntimeError(error)
+        else:
+            raise ValidationError(error)
+
+
+class StabilityV3Adapter(ChatCompletionAdapter):
+    model: str
+    client: Bedrock
+    storage: Optional[FileStorage]
+
+    @classmethod
+    def create(cls, client: Bedrock, model: str, api_key: str):
+        storage: Optional[FileStorage] = create_file_storage(api_key)
+        return cls(
+            client=client,
+            model=model,
+            storage=storage,
+        )
+
+    def _validate_last_message(self, messages: List[Message]):
+        if not messages:
+            raise ValidationError("No messages provided")
+
+        last_message = messages[-1]
+        if last_message.role != Role.USER:
+            raise ValidationError("Last message must be from user")
+        return last_message
+
+    async def compute_discarded_messages(
+        self, params: ModelParameters, messages: List[Message]
+    ) -> DiscardedMessages | None:
+        self._validate_last_message(messages)
+        return list(range(len(messages) - 1))
+
+    async def chat(
+        self,
+        consumer: Consumer,
+        params: ModelParameters,
+        messages: List[Message],
+    ) -> None:
+
+        text_prompt = None
+        image_data = None
+        last_message = self._validate_last_message(messages)
+        # Handle text content
+        match last_message.content:
+            case str(text):
+                text_prompt = text
+            case list():
+                text_parts = []
+
+                for part in last_message.content:
+                    match part:
+                        case MessageContentTextPart(text=text):
+                            text_parts.append(text)
+                        case MessageContentImagePart():
+                            if image_data is not None:
+                                raise ValidationError(
+                                    "Only one input image is supported"
+                                )
```

Comment:
I would have: 
1. collected the resources (both from content parts and attachments) into a `image_resources: List[Resource]`, just like you do with `text_parts`,
2. checked that the list has length no greater than 1, otherwise throw the exception,
3. downloaded the image if there is any

Reply:
Good idea, fixed